src/protocol/reveal_additive_binary.rs

```diff
@@ -19,8 +20,8 @@ pub struct RevealAdditiveBinary {}
 
 impl RevealAdditiveBinary {
     #[allow(dead_code)]
-    pub async fn execute<F: Field>(
-        ctx: ProtocolContext<'_, F>,
+    pub async fn execute<S: SecretShare<F>, F: Field>(
+        ctx: ProtocolContext<'_, S, F>,
         record_id: RecordId,
         input: Fp2,
     ) -> Result<Fp2, BoxError> {
```

Comment:
OK, it's kind of weird that this is the one protocol that appears to be "generic over any kind of secret sharing", and it's a bit concerning because logically speaking, this protocol is _only_ applicable for an *additive secret sharing* (one we don't even have a class for) and because `input` is typed as Fp2.

So everything is a bit weird here. A few ideas:

1. This approach has no path to upgrade to malicious, so I'll probably need to change it to use replicated secret sharing instead of additive.
2. Given that Fields are generic and we have support for Fp2, we should probably just have one, single Reveal protocol and use it for both Fp2 and other cases.
3. This protocol just accepts an Fp2 argument that has no compile-time (or run-time) relationship to the SecretSharing of the protocol context. I'm not sure if that's a problem. I guess it's not... I have to think about this. I think that the SecretSharing (and Field) of the protocol context relate to the malicious security information-theoretic MACs being accumulated. In which case, this is OK. In our protocol we wind up revealing a bool, and that's fine. It's different than the field we are working in. I just need to think about how (if at all) it should affect that MAC.

Reply:
yea this is mostly mechanical change from me to make everything compile and pass the tests. 

> This approach has no path to upgrade to malicious, so I'll probably need to change it to use replicated secret sharing instead of additive.

Yea it is not immediately obvious to me how to protect against cheating helpers inside mod conversion

> Given that Fields are generic and we have support for Fp2, we should probably just have one, single Reveal protocol and use it for both Fp2 and other cases.

I agree, we should make `Reveal` generic over any secret sharing.